lib/create-contentful-api.ts

```diff
@@ -279,352 +306,360 @@ interface GetConfig {
+     * const space = await client.getSpace()
+     * console.log(space)
+     */
+    async function getSpace(): Promise<Space> {
+        return get<Space>({context: 'space', path: ''})
+    }
+
+    /**
+     * Gets a Content Type
+     * @memberof ContentfulClientAPI
+     * @param  {string} id
+     * @return {Promise<Entities.ContentType>} Promise for a Content Type
+     * @example
+     * const contentful = require('contentful')
+     *
+     * const client = contentful.createClient({
+     *   space: '<space_id>',
+     *   accessToken: '<content_delivery_api_key>'
+     * })
+     *
+     * const contentType = await client.getContentType('<content_type_id>')
+     * console.log(contentType)
+     */
+    async function getContentType(id): Promise<ContentType> {
+        return get<ContentType>({
+            context: 'environment',
+            path: `content_types/${id}`
+        })
+    }
+
+    /**
+     * Gets a collection of Content Types
+     * @memberof ContentfulClientAPI
+     * @param  {Object=} query - Object with search parameters. Check the <a href="https://www.contentful.com/developers/docs/javascript/tutorials/using-js-cda-sdk/#retrieving-entries-with-search-parameters">JS SDK tutorial</a> and the <a href="https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters">REST API reference</a> for more details.
+     * @return {Promise<Entities.ContentTypeCollection>} Promise for a collection of Content Types
+     * @example
+     * const contentful = require('contentful')
+     *
+     * const client = contentful.createClient({
+     *   space: '<space_id>',
+     *   accessToken: '<content_delivery_api_key>'
+     * })
+     *
+     * const response = await client.getContentTypes()
+     * console.log(response.items)
+     */
+    async function getContentTypes(query = {}): Promise<ContentTypeCollection> {
+        return get<ContentTypeCollection>({
+            context: 'environment',
+            path: 'content_types',
+            config: createRequestConfig({query: query})
+        })
     }
-    error.details = {
-      type: 'Entry',
-      id: id,
-      environment: getGlobalOptions().environment,
-      space: getGlobalOptions().space
+
+    /**
+     * Gets an Entry
+     * @memberof ContentfulClientAPI
+     * @param  {string} id
+     * @param  {Object=} query - Object with search parameters. In this method it's only useful for `locale`.
+     * @return {Promise<Entities.Entry>} Promise for an Entry
+     * @example
+     * const contentful = require('contentful')
+     *
+     * const client = contentful.createClient({
+     *   space: '<space_id>',
+     *   accessToken: '<content_delivery_api_key>'
+     * })
+     *
+     * const entry = await client.getEntry('<entry_id>')
+     * console.log(entry)
+     */
+    async function getEntry<T>(id, query = {}): Promise<Entry<T>> {
+        if (!id) {
+            throw notFoundError(id)
```

Comment:
Seems a bit weird to throw a not found error that we pass the ID that doesn't exist :)

Reply:
@TimBeyer yeah, but that part is not new :)